hassio/src/supervisor-base-element.ts

```diff
@@ -0,0 +1,69 @@
+import { LitElement, property, PropertyValues } from "lit-element";
+import {
+  fetchHassioHassOsInfo,
+  fetchHassioHostInfo,
+} from "../../src/data/hassio/host";
+import { fetchNetworkInfo } from "../../src/data/hassio/network";
+import { fetchHassioResolution } from "../../src/data/hassio/resolution";
+import {
+  fetchHassioHomeAssistantInfo,
+  fetchHassioInfo,
+  fetchHassioSupervisorInfo,
+} from "../../src/data/hassio/supervisor";
+import { Supervisor } from "../../src/data/supervisor/supervisor";
+import { ProvideHassLitMixin } from "../../src/mixins/provide-hass-lit-mixin";
+import { urlSyncMixin } from "../../src/state/url-sync-mixin";
+
+declare global {
+  interface HASSDomEvents {
+    "supervisor-update": Partial<Supervisor>;
+  }
+}
+
+export class SupervisorBaseElement extends urlSyncMixin(
+  ProvideHassLitMixin(LitElement)
+) {
+  @property({ attribute: false }) public supervisor!: Supervisor;
```

Comment:
It is not always defined, so we should either give it a default value or replace the `!` with an `?`

Reply:
```suggestion
  @property({ attribute: false }) public supervisor?: Supervisor;
```